### DISCUSSION TITLE
Page Rules based on Tag Matches do not work correctly for comment permissions

### DISCUSSION THREAD

**Discussion by StNimmerlein** [requarks/wiki#3598] [Error / Bug Report] [answered]
**Describe the bug**
When adding the permission to read and write comments by a page rule with the "Tag Matches" restraint, the comment box will be displayed on the page but no comments will show and trying to send a comment will result in an error.

**To Reproduce**
1. Create a new User Group and assign it to a user (or use the Guest Group)
2. Create a Page Rule to **Allow** `read:pages`, `read:comments`, `read:assets`, `write:comments` for pages with **Tag Matches** `public`
3. Add the `public` tag to a page
4. Visit this page with said user (or the guest user)
5. You can access and read the page with the current behavior:
   1. You can see the comment input box below.
   2. You see an error message `An unexpected error occured.`
   3. Opening the Browser JS Console, you see `[GraphQL error]: Message: You are not authorized to view comments for this page., Location: [object Object], Path: comments,list`
   4. When trying to write a comment, you get the error `You are not authorized to post a comment on this page.`
   5. If there are already comments on this page, you do not see them

**Expected behavior**
Said user should be able to read and write comments on the page.

**Screenshots**
![Page Rules](https://user-images.githubusercontent.com/16012551/89023052-426d5900-d323-11ea-943d-09800ec0acda.png)
![Admin on page](https://user-images.githubusercontent.com/16012551/89023100-51eca200-d323-11ea-99bb-f45e4d48c894.png)
![Guest on page](https://user-images.githubusercontent.com/16012551/89023128-5c0ea080-d323-11ea-9991-2a2662a8cdb8.png)
![Guest tried to post comment](https://user-images.githubusercontent.com/16012551/89023163-69c42600-d323-11ea-8ffb-4dea979d67e6.png)

**Host Info (please complete the following information):**
 - OS: Linux - Debian (buster) 10 arm (Rasbpian, running on Raspberry Pi 4)
 - Wiki.js version: 2.4.107
 - Database engine: postgres 9.5.21

**Comment by Tunis** [comment]
Hello @NGPixel,

I can confirm this bug with wikijs version : 2.5.287

that's an issue for our use case where we want to allow access to certain group only on matching tag ( only group x has access to tag x )
but overall it makes page-rules with tag unusable ;)

Great work for this wiki, we love it :)

**Comment by hashfunktion** [comment]
I can confirm this issue always in 2.5.289.
When this will work, we can roll out this great wiki to our employees.

Thanks for the great work on Wikijs!

**Comment by MB175** [comment]
Can confirm this bug is also present in version Wiki.js 2.5.290

**Comment by NGPixel** [comment, marked as answer]
Has been addressed in upcoming 2.5.291 update
